### ISSUE TITLE
Y Ratio issue on windows 10 when using a window with decoration

### ISSUE THREAD

**Issue by trashuj** [glfw/glfw#1148]
hi all,
I'm having an issue that might be related.
it happens only on windows 10, windowed mode.
full screen or removing the decoration fixes it. Also it works fine on linux.
i'm using the latest version (I just cloned github)

I attached several screenshots.
basically i'm displaying 5 times the same test texture, so the y position should be 0, 0.2, 0.4, 0.6, 0.8

What i expect is shown in 02_nodeco.png, which is the version without decoration.
The red cross shows where the pointer was when i took the screenshot.
![02_nodeco](https://user-images.githubusercontent.com/9165398/33331214-fb5785aa-d49b-11e7-9de6-63bf4117ccf7.png)

other screenshots show the bug.
00.png is set to 0, as expected, but we can see that the top of the texture ("top") is missing
![00](https://user-images.githubusercontent.com/9165398/33331212-faed4c76-d49b-11e7-9172-58263e0eaf5f.png)

in 02.png, it should be 0.2. instead it's 0.15
![02](https://user-images.githubusercontent.com/9165398/33331213-fb23fe74-d49b-11e7-9c9b-38b3bacf6ebd.png)

in 04.png, it should be 0.4 instead it's 0.36
![04](https://user-images.githubusercontent.com/9165398/33331569-e605605e-d49c-11e7-95ba-c1b61146ae43.png)

in 08.png, it should be 0.8 instead it's 0.79 (somehow catching up)
![08](https://user-images.githubusercontent.com/9165398/33331216-fb8f541c-d49b-11e7-8a2a-331406278c94.png)

here is my test texture for reference.
![dbg_panel](https://user-images.githubusercontent.com/9165398/33332140-68a28c98-d49e-11e7-95fe-612a278c962e.png)

do you have any idea on what could cause this?
any help/hint would be much appreciated.

thanks a lot in advance!

**Comment by ferreiradaselva** [comment]
Some possible problems. Seems like a bad configuration of the GL view matrix, or the quad vertices are `left = -1.0`, `top = -1.0`, `right = 1.0`, `bottom = 1.0` (with 1 possibly being other value, but the center of the quad still being `0.0x0.0`). Or a combination of both, since the quad is stretched.

What are you using to configure the view, the window size or the framebuffer size? If you are using the window size, it might give different results between decorated and undecorated, since decorated include the window frame (border of the window).

**Comment by trashuj** [comment]
thanks a lot for your fast answer!

To be able to use 0-1 I use a vertex shader to convert the coordinates:
	// openGL expects -1.f to 1.f, center = 0,0 but our coords are in [0-1], center being 0.5, 0.5
	// (0, 0) is bottom left
	vec2 pos = _position;
	pos.x = (pos.x * 2.f) - 1.f;
	pos.y = -(pos.y * 2.f) + 1.f;
	gl_Position = vec4(pos, 0.f, 1.f);

the fragment shader is standard:
color_ = vec4(texture(textureSampler, uv).xyz, 1.0f);

to create the window i call:
window_ = glfwCreateWindow(windowInfo.getWidth(), windowInfo.getHeight(), "test", nullptr, nullptr);
glfwMakeContextCurrent(window_);
glViewport(0, 0, windowInfo.getWidth(), windowInfo.getHeight());
witdth/height is by default 800x600

some piece of code i forgot, but to get the correct ratio from the mouse i do:
    // 0 - 1 coords
    Vec2f const mouse01(static_cast<Float>(X) / windowInfo.getWidth(), static_cast<Float>(Y) / windowInfo.getHeight());

what do you mean by "decorated include the window frame"?
like the title bar?
thing is when i change the window size, the ratio changes (even if the title bar has the same size)

**Comment by trashuj** [comment]
I actually just realized that i have the problem on X axis as well (and it shows on the screenshots).
where the pic should start at 0.1, the mouse position is actually 0.94.

@ferreiradaselva if you're right about the decoration, how can i get the "real" window size without deco?

**Comment by ferreiradaselva** [comment]
> what do you mean by "decorated include the window frame"?
> like the title bar?
Yes. `glfwGetWindowSize` includes the size of the frame (titlebar and borders). `glfwGetFramebufferSize` don't.

> thing is when i change the window size, the ratio changes (even if the title bar has the same size)

You should also reset the viewport in the FramebufferSizeCallback (you can set this callback using `glfwSetFramebufferSizeCallback`). The viewport is not automatically resized when you resize the window.

**Comment by trashuj** [comment]
Actually glfwGetWindowSize and glfwGetFramebufferSize give me the same values.
Using glfwSetFramebufferSizeCallback instead of glfwSetWindowSizeCallback has the same behavior.

What i do in the callback:
void windowResizeCallback(GLFWwindow * window, int width, int height)
{
    data::Window & windowInfo = engine->getGameData().getDisplay().getWindow();
    windowInfo.setWidth(width);
    windowInfo.setHeight(height);
    glViewport(0, 0, width, height);
}

**Comment by ferreiradaselva** [comment]
> Actually glfwGetWindowSize and glfwGetFramebufferSize give me the same values.

That's weird.

```c
void windowResizeCallback(GLFWwindow * window, int width, int height)
{
data::Window & windowInfo = engine->getGameData().getDisplay().getWindow();
windowInfo.setWidth(width);
windowInfo.setHeight(height);
glViewport(0, 0, width, height);
}
```

Is this a framebuffer size callback or a window size callback (it's named as if it was a window size callback)?

**Comment by trashuj** [comment]
I just copy pasted the code i was using with window size callback, so i'm
using this for both (only framebuffersize callback is called now)


On Dec 3, 2017 10:24, "pengo" <notifications@github.com> wrote:

Actually glfwGetWindowSize and glfwGetFramebufferSize give me the same
values.

That's weird.

void windowResizeCallback(GLFWwindow * window, int width, int height)
{
data::Window & windowInfo = engine->getGameData().getDisplay().getWindow();
windowInfo.setWidth(width);
windowInfo.setHeight(height);glViewport(0, 0, width, height);
}

Is this a framebuffer size callback or a window size callback (it's named
as if it was a window size callback)?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/glfw/glfw/issues/1148#issuecomment-348735315>, or mute
the thread
